internal/rdb/rdb.go

```diff
@@ -112,8 +114,11 @@ func (r *RDB) Enqueue(ctx context.Context, msg *base.TaskMessage) error {
 	if err != nil {
 		return errors.E(op, errors.Unknown, fmt.Sprintf("cannot encode message: %v", err))
 	}
-	if err := r.client.SAdd(ctx, base.AllQueues, msg.Queue).Err(); err != nil {
-		return errors.E(op, errors.Unknown, &errors.RedisCommandError{Command: "sadd", Err: err})
+	if _, ok := r.m.Load(msg.Queue); !ok {
```

Comment:
Maybe this for atomicity:
```suggestion
	if previous, _ := r.m.Swap(msg.Queue, true); previous == nil {
```

Since we have many nodes, we probably don't need to check the error.